lib/cylc/job_host.py

```diff
@@ -183,8 +183,8 @@ def unlink_suite_contact_files(self, reg):
         # Wait for commands to complete for a max of 10 seconds
         timeout = time() + 10.0
         while procs and time() < timeout:
-            for user_at_host, (cmd, proc) in procs.items():
-                if not proc.poll():
+            for user_at_host, (cmd, proc) in procs.copy().items():
```

Comment:
There is a similar discussion regarding this in #2086. Since we are removing items from the dict `procs`, it is safer to iterate a copy of it.